src/Orleans/GrainDirectory/MultiClusterStatus.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Orleans.GrainDirectory
+{
+    /// <summary>
+    /// Status of a directory entry with respect to multi-cluster registration
+    /// </summary>
+    internal enum MultiClusterStatus : byte
+    {
+        /// <summary>
+        /// Registration is owned by this cluster.
+        /// </summary>
+        Owned,
+
+        /// <summary>
+        /// Failed to contact one or more clusters while registering, so may be a duplicate.
+        /// </summary>
+        Doubtful,
+
+        /// <summary>
+        /// Cached reference to a registration owned by a remote cluster.
+        /// </summary>
+        Cached,
+
+        /// <summary>
+        /// The cluster is in the process of checking remote clusters for existing registrations.
+        /// </summary>
+        RequestedOwnership,
+
+        /// <summary>
+        /// The cluster lost a race condition.
+        /// </summary>
+        RaceLoser,
```

Comment:
RaceLoser is needed to block concurrent racing request. It is set by a remote request arriving on a node that is in RequestedOwnership state, to tell that request that some other one with higher priority. I don't think any of these states can be collapsed further, we studied this protocol quite carefully.

Arguably, these names are tightly related to the global single instance protocol and do not make general sense for other protocols. We could consider hiding the names in a less prominent place, e.g. just reserve one uninterpreted byte.